File: src/main/java/yuma140902/yumalib/api/util/NameProvider.java
package yuma140902.yumalib.api.util;

import javax.annotation.Nonnull;
import javax.annotation.Nullable;

/**
 * ドメインなしの文字列を受け取りドメイン付きの文字列を返す
 */
public class NameProvider {
	
	public final String TEXTURE_DOMAIN;
	public final String UNLOCALIZED_DOMAIN;
	
	public NameProvider(String textureDomain, String unlocalizedDomain) {
		this.TEXTURE_DOMAIN = textureDomain;
		this.UNLOCALIZED_DOMAIN = unlocalizedDomain;
	}
	
	public String domainedTexture(String name) {
		return TEXTURE_DOMAIN + ":" + name;
	}
	
	public String domainedUnlocalized(String name) {
		return UNLOCALIZED_DOMAIN + "." + name;
	}
	
	
	@Nonnull
	public String[] domainedTextures(String... names) {
		return _domainedTextures(TEXTURE_DOMAIN, names);
	}
	
	@Nonnull
	private static String[] _domainedTextures(String textureDomain, @Nullable String... names) {
		if(names == null) return new String[0];
		
		String[] ret = new String[names.length];
		for (int i = 0; i < names.length; i++) {
			ret[i] = textureDomain + ":" + names[i];
		}
		return ret;
	}
}

File: src/main/scala/yuma140902/yumalib/api/util/NameProvider.scala
package yuma140902.yumalib.api.util

/**
 * ドメインなしの文字列を受け取りドメイン付きの文字列を返す
 */
case class NameProvider(textureDomain: String, unlocalizedDomain: String) {
  def domainedUnlocalized(name: String): String = unlocalizedDomain + "." + name;

  def domainedTextures(names: Name*): Seq[NameWithModID] = names.map(domainedTexture)

  def domainedTexture(name: Name): NameWithModID = NameWithModID(textureDomain + ":" + name.str)

  @deprecated("", "2.4.1")
  def domainedTextures_(names: String*): Seq[String] = names.map(domainedTexture)

  @deprecated("", "2.4.1")
  def domainedTexture(name: String): String = textureDomain + ":" + name

}
